### PR TITLE
Document reserved routes

### PR DESCRIPTION
**Pull request by Fryuni**
#### Description

Document the routes used internally by Astro that will override any user and integration routes.

- https://github.com/withastro/astro/pull/12263#pullrequestreview-2381408289

### CODE REVIEW

File: src/content/docs/en/guides/routing.mdx [hunk 1]
```diff
@@ -392,6 +393,14 @@ Given the example above, here are a few examples of how the rules will match a r
 - `pages/posts/[...slug].astro` - Will build `/posts/1/2`, `/posts/a/b/c`, etc. But not `/posts/create`, `/posts/1`, `/posts/abc`, etc.
 - `pages/[...slug].astro` - Will build `/abc`, `/xyz`, `/abc/xyz`, etc. But not `/posts/create`, `/posts/1`, `/posts/abc`, , etc.
 
+### Reserved routes
+
+Some Astro features require exposing routes to be accessible by the clients. Those routes take priority over any user-defined or integration-defined routes.
+
+- `_astro/`: Serves all the static assets to the client. Including CSS documents, bundled client scripts, optimized images and any Vite asset;
+- `_server_islands/`: Serves the dynamic components deferred into a [Server Island](/en/reference/configuration-reference/#experimentalserverislands);
+- `_actions/`: Serves the defined [Actions](/en/guides/actions/).
```
Comment (sarah11918) [inline src/content/docs/en/guides/routing.mdx:402]:
```suggestion
- `_astro/`: Serves all of the static assets to the client, including CSS documents, bundled client scripts, optimized images, and any Vite assets.
- `_server_islands/`: Serves the dynamic components deferred into a [server island](/en/reference/configuration-reference/#experimentalserverislands);
- `_actions/`: Serves any defined [actions](/en/guides/actions/).
```

File: src/content/docs/en/guides/routing.mdx [hunk 2]
```diff
@@ -392,6 +393,14 @@ Given the example above, here are a few examples of how the rules will match a r
 - `pages/posts/[...slug].astro` - Will build `/posts/1/2`, `/posts/a/b/c`, etc. But not `/posts/create`, `/posts/1`, `/posts/abc`, etc.
 - `pages/[...slug].astro` - Will build `/abc`, `/xyz`, `/abc/xyz`, etc. But not `/posts/create`, `/posts/1`, `/posts/abc`, , etc.
 
+### Reserved routes
+
+Some Astro features require exposing routes to be accessible by the clients. Those routes take priority over any user-defined or integration-defined routes.
```
Comment (sarah11918) [inline src/content/docs/en/guides/routing.mdx:398]:
```suggestion
Internal routes take priority over any user-defined or integration-defined routes as they are required for Astro features to work. The following are Astro's reserved routes:
```

**Review by sarah11918 — APPROVED** [review]
looks fabulous @Fryuni ! Left two tiny suggestions for the exact wording, but as I can see the feature has already merged and been released, this is good for you to merge whenenever you're happy with it!

**Comment by Fryuni** [comment]
> as I can see the feature has already merged and been released

Just to clarify, the linked PR does not add this behavior. This behavior already applied, but it wasn't documented (even on the code), and it accidentally broke at one point. The linked PR is the fix for that where we realized it wasn't documented.

**Pull request merged**